packages/@okta/vuepress-site/docs/guides/oin-oidc-guide/protocol-level-requirements/index.md

```diff
@@ -2,52 +2,98 @@
+
+### Authorization Code flow with PKCE
+
+f you are building a Single Page Application (SPA) for the OIN, then Okta recommends using the Authorization Code flow with a Proof Key for Code Exchange (PKCE) to control the access between your application and a resource server. See our [OAuth 2.0 overview](/docs/concepts/oauth-openid/#authorization-code-with-pkce) for more information on the authorization code with PKCE flow, including why to use it.
+
+Just like with the regular Authorization Code flow, your app starts by redirecting the end user's browser to your [Authorization Server's](/docs/concepts/auth-servers/) `/authorize` endpoint. However, in this instance, you also have to pass along a code challenge.
+
+Your first step is to generate a code verifier and challenge:
+
+* Code verifier: Random URL-safe string with a minimum length of 43 characters.
+* Code challenge: Base64 URL-encoded SHA-256 hash of the code verifier.
+
+You need to add code in your SPA app to create the code verifier and code challenge.
+
+The Authorization Code flow with PKCE looks like this for an OIN app:
+
+* Your end user (also known as the Resource Owner) opens the SPA app in their browser so that they can access and work with your application (the Client). The Client requires information from various back-end infrastructure elements (the Resource Server), usually an API or service. By clicking on a sign-in link, the end user grants authorization to your application and initiates the Authorization Code flow with PKCE.
+* Your client application prepares an authentication request to authenticate the end user. This request is composed of specific required parameters and is sent to the Okta `/authorize` endpoint with a code challenge:
+  * `client_id` &mdash; this value is the Okta client ID created for you when you made your Okta app integration and is available in your Okta app integration settings
+  * `redirect_uri` &mdash; the location where the response is sent. This value, which must start with `https://`, is set in your Okta app integrations settings
+  * `response_type=code` &mdash; this informs Okta (as the authorization server) that you want to get back an access token and an ID token in exchange for the authorization code
+  * `state` &mdash; an arbitrary value returned in the token. The client application can use it to remember the state of its interaction with the end user at the time of the authentication call
+  * `scope` &mdash; must contain `openid` to indicate that you want to get back an ID token from the `/token` endpoint
+  * `code_challenge` &mdash; the PKCE code challenge you previously generated
+  * `code_challenge_method` &mdash; the hash method used to generate the challenge value; this value is always `S256`
```

Comment:
```suggestion
  * `code_challenge_method` &mdash; the hash method used to generate the challenge value. This value is always `S256`.
```

Reply:
OK - clarified use of periods with Susan in slack chat.